x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1401,6 +1430,16 @@ private EventFilterPolicy changeIndicesFilter(List<String> filtersList) {
             return l.stream().map(f -> f.isEmpty() ? "//" : f).collect(Collectors.toList());
         }
 
+        private boolean ignorePrivilegesPredicateTest(String action) {
+            Collection<String> privileges = null;
+            if (AuthorizationService.isIndexAction(action)) {
+                privileges = IndexPrivilege.findPrivilegesThatGrant(action);
+            } else if (ClusterPrivilegeResolver.isClusterAction(action)) {
+                privileges = ClusterPrivilegeResolver.findPrivilegesThatGrant(action, null, null);
+            }
+            return privileges != null ? privileges.stream().anyMatch((s) -> ignorePrivilegesPredicate.test(s)) : false;
```

Comment:
I see that you've added some caching.
But, if it is a caching from action name to privilege names, then it should be a more global one, and not at the policy level. If it is at the policy level the cache can map directly to a boolean saying if the policy matches the action.
I would say that both can work together.
If possible, make them bounded, so that if we have a bug we experience poor performance rather than grow the memory unpredictably.